pyvista/core/utilities/geometric_objects.py

```diff
@@ -363,6 +364,354 @@ def Sphere(
+    Generate a solid hemi-sphere to see the internal structure.
+
+    >>> isinstance(solid_sphere, pv.UnstructuredGrid)
+    True
+    >>> partial_solid_sphere = pv.SolidSphere(
+    ...     start_theta=180, end_theta=360
+    ... )
+    >>> partial_solid_sphere.plot(show_edges=True)
+
+    To see the cell structure inside the solid sphere,
+    only 1/4 of the sphere is generated. The cells exploded
+    and colored by radial position.
+
+    >>> partial_solid_sphere = pv.SolidSphere(
+    ...     start_theta=180,
+    ...     end_theta=360,
+    ...     start_phi=0,
+    ...     end_phi=90,
+    ...     radius_resolution=5,
+    ...     theta_resolution=8,
+    ...     phi_resolution=8,
+    ... )
+    >>> partial_solid_sphere["cell_radial_pos"] = np.linalg.norm(
+    ...     partial_solid_sphere.cell_centers().points, axis=-1
+    ... )
+    >>> partial_solid_sphere.explode(1).plot()
+
+    """
+    if radius is None:
+        radius = np.linspace(inner_radius, outer_radius, radius_resolution)
+    if theta is None:
+        theta = np.linspace(start_theta, end_theta, theta_resolution)
+    if phi is None:
+        phi = np.linspace(start_phi, end_phi, phi_resolution)
+
+    if radius[0] < 0:
+        raise ValueError("minimum radius cannot be negative")
+    if theta[0] < 0:
+        raise ValueError("minimum theta cannot be negative")
+    if phi[0] < 0:
+        raise ValueError("minimum phi cannot be negative")
+    if theta[-1] > 360:
+        raise ValueError("maximum theta cannot be > 360")
+    if phi[-1] > 180:
+        raise ValueError("maximum phi cannot be > 180")
+
+    # Hereafter all degrees are in radians
+    theta = np.deg2rad(theta)
+    phi = np.deg2rad(phi)
+
+    nr = len(radius)
+    ntheta = len(theta)
+    nphi = len(phi)
+
+    if nr < 2:
+        raise ValueError("radius_resolution must be 2 or more")
+    if ntheta < 2:
+        raise ValueError("theta_resolution must be 2 or more")
+    if nphi < 2:
+        raise ValueError("phi_resolution must be 2 or more")
+
+    def _spherical_to_cartesian(r, phi, theta):
+        """Convert spherical coordinate sequences to a ``(n,3)`` cartesian coordinate array.
+
+        Parameters
+        ----------
+        r : sequence[float]
+            Ordered sequence of floats of radii.
+        phi : sequence[float]
+            Ordered sequence of floats for phi direction.
+        theta : sequence[float]
+            Ordered sequence of floats for theta direction.
+
+        Returns
+        -------
+        np.ndarray
+            ``(n, 3)`` cartesian coordinate array.
+
+        """
+        r, phi, theta = np.meshgrid(r, phi, theta, indexing='ij')
+        x = r * np.sin(phi) * np.cos(theta)
+        y = r * np.sin(phi) * np.sin(theta)
+        z = r * np.cos(phi)
+        return np.vstack((x.ravel(), y.ravel(), z.ravel())).transpose()
+
+    points = []
+
+    npoints_on_axis = 0
+
+    if radius[0] == 0.0:
```

Comment:
How about taking into account the error of floating point number float including other parts?

Reply:
This is a good point, but I'm not sure whether we should police it here or not.  What if the user really does want a very small number here that is different than 0?

Reply:
I see. Given that use case, it is better to judge strictly by equals. Let's make a strict judgment first and see how users react.

Reply:
The most common use case is that this is triggered with the defaults for each of radius, phi, and theta, which we know that they are identically equal.  In the case that we get user feedback that they need to specify whether they intended to have an origin with a close value of 0, we could add in a flag and/or a threshold parameter.  A flag could also be useful to specify that they user doesn't want to consider theta=0 and theta=360 to be a seamless mesh.  I agree on your assessment that getting user feedback is needed at this time.

Reply:
What about `np.isclose(radius[0], 0)`? [np.isclose docs.](https://numpy.org/doc/stable/reference/generated/numpy.isclose.html). This way if the user gets a radius of zero as a result of some computation that should have been zero but was different by a small amount of numerical quantization, it will still be considered to be zero. And anything the user is doing can't distinguish between 0 and whatever value they're using anyways. `np.isclose` is going to ask as a threshold parameter, but I would argue the value of that threshold is not arbitrary. This is the use case for `np.isclose`.

I agree there could be a range of other behaviors users might want though.